src/main/Core/ImageGenerator/Linux/LinuxAppIconExtractor.ts

```diff
@@ -271,15 +273,52 @@ export class LinuxAppIconExtractor implements FileIconExtractor {
     }
 
     private async getIconThemeName(): Promise<string> {
-        const currentDesktopEnvironment = this.environmentVariablePRovider.get("XDG_SESSION_DESKTOP");
-
-        const iconThemeNameMap: Record<"cinnamon" | "gnome", () => Promise<string>> = {
-            cinnamon: () =>
-                this.commandlineUtility.executeCommand("gsettings get org.cinnamon.desktop.interface icon-theme"),
-            gnome: () => this.commandlineUtility.executeCommand("gsettings get org.gnome.desktop.interface icon-theme"),
+        const iconThemeNameMap: Record<LinuxDesktopEnvironment, () => Promise<string>> = {
```

Comment:
I suggest using just strings here to untangle the `LinuxAppIconGenerator` from the application search extension.

```suggestion
        const iconThemeNameMap: Record<string, () => Promise<string>> = {
```